File: Tyche.iOS/Bet/Sources/Bet/Domain/Bet.swift
import Core

public struct Bet {
    let poolId: String
    let gamblerId: String
    let matchId: String
    let homeTeamBet: BetScore
    let awayTeamBet: BetScore
}

extension Bet {
    func toBetRequest() -> BetRequest {
        BetRequest(
            poolId: self.poolId,
            gamblerId: self.gamblerId,
            matchId: self.matchId,
            homeTeamBet: self.homeTeamBet.value,
            awayTeamBet: self.awayTeamBet.value
        )
    }
}

File: Tyche.Android/data/bet/src/main/java/com/felipearpa/data/bet/domain/Bet.kt
package com.felipearpa.data.bet.domain

import com.felipearpa.tyche.core.type.BetScore

data class Bet(
    val poolId: String,
    val gamblerId: String,
    val matchId: String,
    val homeTeamBet: BetScore,
    val awayTeamBet: BetScore
)